zookeeper/source/zk_nn_client.cc

```diff
@@ -837,6 +842,19 @@ bool ZkNnClient::create_file_znode(const std::string &path,
                    << error_code;
         return false;
       }
+      // grant the client the lease
+      if (!zk->create(LeaseZookeeperPath(path) + '/' +
+                        client_name, ZKWrapper::EMPTY_VECTOR,
+                      error_code, false)) {
+        LOG(ERROR) << "[create_file_znode] ZK create lease client path failed"
+        << error_code;
+        return false;
+      }
+      // set the timestamp for the client that created the file
+      RenewLeaseRequestProto req;
+      req.set_clientname(client_name);
+      RenewLeaseResponseProto res;
+      renew_lease(req, res);
```

Comment:
This looks a messy workaround for creating a lease. Do we not have a "helper function" for creating leases? I might be wrong about there being a cleaner way though; just wondering.